src/main/java/seedu/address/logic/commands/EditAppointmentCommand.java

```diff
@@ -53,33 +53,40 @@
 
     public static final String MESSAGE_EDIT_APPOINTMENT_SUCCESS = "Edited Appointment: %1$s";
     public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
-    public static final String MESSAGE_DUPLICATE_APPOINTMENT = "This  already exists in the Appointment list.";
-
+    public static final String MESSAGE_APPOINTMENT_CONFLICT = "This appointment will result in conflicts.";
+    private final Index index;
     private final EditAppointmentDescriptor editAppointmentDescriptor;
 
     /**
+     * @param index of the appointment in the filtered appointment list to edit
      * @param editAppointmentDescriptor details to edit the appointment with
      */
-    public EditAppointmentCommand(EditAppointmentDescriptor editAppointmentDescriptor) {
-        requireAllNonNull(editAppointmentDescriptor);
+    public EditAppointmentCommand(Index index, EditAppointmentDescriptor editAppointmentDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editAppointmentDescriptor);
+        this.index = index;
         this.editAppointmentDescriptor = new EditAppointmentDescriptor(editAppointmentDescriptor);
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        List<Appointment> appointmentList = model.getFilteredAppointmentList();
-
-        // patient index == appointment index
-        if (editAppointmentDescriptor.patientIndex.getZeroBased() >= appointmentList.size()) {
+        List<Person> displayedPatientRecords = model.getFilteredPersonList();
+        if (editAppointmentDescriptor.patientIndex.getZeroBased() >= displayedPatientRecords.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_PATIENT_DISPLAYED_INDEX);
         }
+        Person patient = displayedPatientRecords.get(editAppointmentDescriptor.patientIndex.getZeroBased());
```

Comment:
this code would not work if patientIndex didn't exist. Should handle it like lines 104-106: ie. something like
```
Person patient = editAppointmentDescriptor.getPatientIndex().isPresent()
        ? displayedPatientRecords.get(editAppointmentDescriptor.patientIndex.getZeroBased())
        : appointmentToEdit.getPatient();
```

Reply:
hi, i think its handled in the previous 3 lines? i wrote:

List<Person> displayedPatientRecords = model.getFilteredPersonList();
        if (editAppointmentDescriptor.patientIndex.getZeroBased() >= displayedPatientRecords.size()) {
            throw new CommandException(Messages.MESSAGE_INVALID_PATIENT_DISPLAYED_INDEX);
        }
        Person patient = displayedPatientRecords.get(editAppointmentDescriptor.patientIndex.getZeroBased());

Reply:
i get what you mean. i try to change

Reply:
The logic flow should be:
IF patientIndex is provided -> user wants to change patient for the appointment {
    patient = patient at patientIndex
} else {
    patient = same patient from appointment
}

Reply:
yup, have made the changes and pushed again. Thank u